### PR TITLE
Bug 1162174 – Allow Third Party Password Manager Extensions.

### PR DESCRIPTION
**Pull request by jhugman**
Work in progress patch rebasing https://github.com/mozilla/firefox-ios/pull/510 before refactoring.

Fails at linking, despite a clean and rm DerivedData.

### CODE REVIEW

**Comment by jhugman** [comment]
Continuing discussion from https://github.com/AgileBits/onepassword-app-extension/issues/277 .

While we're waiting for LastPass to come back to us, a thought strikes me: if LastPass and our schedules don't coincide, would you accept a PR to open up access to the `kUTTypeAppExtension`* and `OPWebView`* constants?

That gives me a small amount of code I need to re-write in Swift to construct multiple versions of the `NSExtensionItem`s without multiple calls into the webview's JS.

It won't open your API up to anything onerous for you to support since it is already discrete function call which needs a well defined set of pre-conditions and whose output is in the actual format which Containing Apps need to implement.

But it does help me give OnePassword users the premium features of OnePassword, while not breaking either `onepassword-app-extension` or shutting out LastPass users.

Once LastPass is up-to-date, that swift code can be reconsidered.

I have other options, but all of them I like less than this option.

File: Cartfile [hunk 1]
```diff
@@ -13,3 +13,4 @@ github "st3fan/SWXMLHash"           "xcode-7.0"        # So we get Swift 2.0 fix
 github "DaveWoodCom/XCGLogger"      "swift_2.0"        # Previously: swift_1.2
 github "sleroux/google-breakpad-ios" "master"
 github "adjust/ios_sdk"             "v4.3.0"
+github "AgileBits/onepassword-extension" "add-framework-support"
```
Comment (st3fan) [inline Cartfile]: Can you pin this to their 1.6.4 tag? Which is the most recent version in the add-framework-support branch?

Comment (st3fan) [inline Cartfile]: Or is this pinned at that commit for a specific reason?

Comment (radazzouz) [inline Cartfile]: @st3fan @jhugman, the [App Extension API](https://github.com/AgileBits/onepassword-app-extension) supports Carthage on the `add-framework-support`. Which is being kept in sync with master. We strongly recommend that you point to its latest commit, `8afe2fad841c0439c4db6f8e2ac4d4c7e1e7a4bd` (the merge commit right after the `1.6.4`) or to the `1.6.4` tag itself.

**Comment by raindev** [comment]
In what version of the app is it planned to be released to AppStore?

**Pull request merged**